From 12fe6f38fa3addb4e9e49873a075b86435e8b301 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Fri, 16 Jan 2026 22:57:37 +0100
Subject: [PATCH] Fix build of Plasmoid against libplasma from beta 6.5.90

Change f56f5eb4aea5f0604636ee5afa791d7672290612 broke the ABI and API of
`Theme::color()`.

This code is just using the default value for the incompatible parameter so
it is possible to allow compilation before and after this change by simply
removing the argument.
---
 plasmoid/lib/syncthingapplet.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plasmoid/lib/syncthingapplet.cpp b/plasmoid/lib/syncthingapplet.cpp
index 595cf937..b38d2f79 100644
--- a/plasmoid/lib/syncthingapplet.cpp
+++ b/plasmoid/lib/syncthingapplet.cpp
@@ -225,7 +225,7 @@ void SyncthingApplet::initEngine(QObject *object)
     if (!engine) {
         return;
     }
-    const auto color = m_theme.color(Plasma::Theme::TextColor, Plasma::Theme::NormalColorGroup);
+    const auto color = m_theme.color(Plasma::Theme::TextColor);
     m_imageProvider
         = new QtForkAwesome::QuickImageProvider(QtForkAwesome::Renderer::global(), color); // using global renderer for system icons override
     connect(engine, &QObject::destroyed, this, &SyncthingApplet::handleImageProviderDestroyed); // engine has ownership over image provider
@@ -721,7 +721,7 @@ void SyncthingApplet::handleThemeChanged()
         m_iconManager.setPalette(palette);
         setBrightColors(isPaletteDark(palette));
         if (m_imageProvider) {
-            m_imageProvider->setDefaultColor(m_theme.color(Plasma::Theme::TextColor, Plasma::Theme::NormalColorGroup));
+            m_imageProvider->setDefaultColor(m_theme.color(Plasma::Theme::TextColor));
         }
         emit faUrlChanged(m_faUrl = QStringLiteral("image://fa/"));
     });
